omero/analysis_scripts/Kymograph_Analysis.py

```diff
@@ -81,128 +79,123 @@ def processImages(conn, scriptParams):
-                        if micronsPerSec:
-                            speed = dxPerY * micronsPerSec
-                            avgSpeed = avXperY * micronsPerSec
-                        tableData += "\n"
-                        tableData += ",".join(
-                            [str(x) for x in (y1, x1, y2, x2, dy, dx, dxPerY,
-                                              speed, avXperY, avgSpeed)])
+                        avg_speed = ""
+                        if microns_per_sec:
+                            speed = dx_per_y * microns_per_sec
+                            avg_speed = av_x_per_y * microns_per_sec
+                        table_data += "\n"
+                        table_data += ",".join(
+                            [str(x) for x in (y1, x1, y2, x2, dy, dx, dx_per_y,
+                                              speed, av_x_per_y, avg_speed)])
 
         # write table data to csv...
-        if len(tableData) > 0:
-            tableString = "Image ID:, %s," % image.getId()
-            tableString += "Name:, %s" % image.getName()
-            tableString += "\nsecsPerPixelY: %s" % secsPerPixelY
-            tableString += '\nmicronsPerPixelX: %s' % micronsPerPixelX
-            tableString += "\n"
-            tableString += colNames
-            tableString += tableData
-            print tableString
-            csvData.append(tableString)
-        else:
-            print "Found NO lines or polylines to analyze for Image"
+        if len(table_data) > 0:
+            table_string = "Image ID:, %s," % image.getId()
+            table_string += "Name:, %s" % image.getName()
+            table_string += "\nsecsPerPixelY: %s" % secs_per_pixel_y
+            table_string += '\nmicronsPerPixelX: %s' % microns_per_pixel_x
+            table_string += "\n"
+            table_string += col_names
+            table_string += table_data
+            csv_data.append(table_string)
 
     iids = [str(i.getId()) for i in images]
-    toLinkCsv = [i.getId() for i in images if i.canAnnotate()]
-    csvFileName = 'kymograph_velocities_%s.csv' % "-".join(iids)
-    csvFile = open(csvFileName, 'w')
-    try:
-        csvFile.write("\n \n".join(csvData))
-    finally:
-        csvFile.close()
+    to_link_csv = [i.getId() for i in images if i.canAnnotate()]
+    csv_file_name = 'kymograph_velocities_%s.csv' % "-".join(iids)
+    with open(csv_file_name, 'w') as csv_file:
+        csv_file.write("\n \n".join(csv_data))
 
-    fileAnn = conn.createFileAnnfromLocalFile(csvFileName, mimetype="text/csv")
-    faMessage = "Created Line Plot csv (Excel) file"
+    file_ann = conn.createFileAnnfromLocalFile(csv_file_name,
+                                               mimetype="text/csv")
+    fa_message = "Created Line Plot csv (Excel) file"
 
     links = []
-    if len(toLinkCsv) == 0:
-        faMessage += " but could not attach to images."
-    for iid in toLinkCsv:
-        print "linking csv to Image: ", iid
```

Comment:
I have removed all the print than are there for "debugging" while writing the tests
The relevant "prints" were already added to the log